crates/matrix-sdk/src/room/mod.rs

```diff
@@ -2334,6 +2401,25 @@ impl Receipts {
     }
 }
 
+/// [Parent space](https://spec.matrix.org/v1.8/client-server-api/#mspaceparent-relationships)
+/// listed by a room, possibly validated by checking the space's state.
+#[derive(Debug)]
+pub enum ParentSpace {
+    /// The room recognizes the given room as its parent, and the parent
+    /// recognizes it as its child.
+    Reciprocal(Room),
+    /// The room recognizes the given room as its parent, but the parent does
+    /// not recognizes it as its child. However, a member of the parent has
+    /// a sufficient power level to do so.
+    WithPowerlevel(Room),
+    /// The room recognizes the given room as its parent, but the parent does
+    /// not recognizes it as its child.
+    Illegitimate(Room),
+    /// The room recognizes the given id as its parent room, but we cannot check
+    /// whether the parent does not recognizes it as its child.
```

Comment:
```suggestion
    /// The room recognizes the given id as its parent room, but we cannot check
    /// whether the parent recognizes it as its child.
```

has the same meaning and is easier to read.